src/sql/parts/modelComponents/componentBase.ts

```diff
@@ -222,17 +224,30 @@ export abstract class ContainerBase<T> extends ComponentBase {
 	}
 
 	/// IComponent container-related implementation
-	public addToContainer(componentDescriptor: IComponentDescriptor, config: any): void {
+	public addToContainer(componentDescriptor: IComponentDescriptor, config: any, index?: number): void {
 		if (this.items.some(item => item.descriptor.id === componentDescriptor.id && item.descriptor.type === componentDescriptor.type)) {
 			return;
 		}
-		this.items.push(new ItemDescriptor(componentDescriptor, config));
+		if (index !== undefined && index !== null && index >= 0 && index < this.items.length) {
+			this.items.splice(index, 0, new ItemDescriptor(componentDescriptor, config));
+		} else {
+			this.items.push(new ItemDescriptor(componentDescriptor, config));
+		}
 		this.modelStore.eventuallyRunOnComponent(componentDescriptor.id, component => component.registerEventHandler(event => {
 			if (event.eventType === ComponentEventType.validityChanged) {
 				this.validate();
 			}
 		}));
 		this._changeRef.detectChanges();
+		return;
+	}
+
+	public removeFromContainer(componentDescriptor: IComponentDescriptor): void {
```

Comment:
Minor: this is most commonly a boolean and returns false if not found. This can be helpful in "remove if exists" scenarios